x-pack/plugins/maps/public/classes/layers/new_vector_layer_wizard/wizard.tsx

```diff
@@ -0,0 +1,124 @@
+import { RenderWizardArguments } from '../layer_wizard_registry';
+import { VectorLayer } from '../vector_layer';
+import { ESSearchSource } from '../../sources/es_search_source';
+import { ADD_LAYER_STEP_ID } from '../../../connected_components/add_layer_panel/view';
+import { getIndexNameFormComponent } from '../../../kibana_services';
+
+interface State {
+  indexName: string;
+  indexError: string;
+  indexingTriggered: boolean;
+  indexPatternId: string;
+}
+
+export class NewVectorLayerEditor extends Component<RenderWizardArguments, State> {
+  private _isMounted: boolean = false;
+
+  state: State = {
+    indexName: '',
+    indexError: '',
+    indexingTriggered: false,
+    indexPatternId: '',
+  };
+
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  async componentDidUpdate() {
+    if (this.props.currentStepId === ADD_LAYER_STEP_ID && !this.state.indexingTriggered) {
+      this.setState({ indexingTriggered: true });
+      await this._createNewIndex();
+      if (!this._isMounted) {
+        return;
+      }
+      this.props.advanceToNextStep();
+    }
+  }
+
+  _createNewIndex = async () => {
+    const { indexPatternId } = await createNewIndexAndPattern(this.state.indexName);
+    if (!this._isMounted) {
+      return;
+    }
+    // Creates empty layer
+    const sourceDescriptor = ESSearchSource.createDescriptor({
+      indexPatternId,
+      geoField: 'coordinates',
+      filterByMapBounds: false,
+    });
+    const layerDescriptor = VectorLayer.createDescriptor(
+      { sourceDescriptor },
+      this.props.mapColors
+    );
+    this.props.previewLayers([layerDescriptor]);
+  };
+
+  _onIndexChange = (indexName: string, indexError?: string) => {
+    this.setState(
+      {
+        indexName,
+        ...(indexError ? { indexError } : { indexError: '' }),
+      },
+      () => {
+        const { enableNextBtn, disableNextBtn } = this.props;
+        if (this.state.indexName && !this.state.indexError) {
+          enableNextBtn();
+        } else {
+          disableNextBtn();
+        }
+      }
+    );
+  };
+
+  render() {
+    const IndexNameForm = getIndexNameFormComponent();
+    return (
+      <EuiPanel>
+        <>
+          <EuiEmptyPrompt
+            title={
+              <h4>
+                {i18n.translate('xpack.maps.layers.newVectorLayerWizard.drawVectorShapes', {
+                  defaultMessage: 'Draw shapes',
+                })}
+              </h4>
+            }
+            body={
+              <Fragment>
+                <p>
+                  {i18n.translate(
+                    'xpack.maps.layers.newVectorLayerWizard.vectorEditorDescription',
+                    {
+                      defaultMessage: `Use the editor on the left to draw your points and shapes.`,
+                    }
+                  )}
+                </p>
+              </Fragment>
+            }
+          />
+          <IndexNameForm
```

Comment:
Pass in the below props give users feedback when name is getting validated
```
onIndexNameValidationEnd={this.props.enableNextBtn}
onIndexNameValidationStart={this.props.disableNextBtn}
```

Reply:
I thinks this logic needs some rethinking on the `IndexNameForm` side but not sure it fits in this PR. Using that logic above enables the button even when there are errors. Even adding an if statement to check for errors on the callback runs into timing issues that leave the button active with errors in place:
```
() => {
  if (this.state.indexName && !this.state.indexError) {
    this.props.enableNextBtn();
  }
}
```